src/app/loadout-drawer/loadout-drawer-reducer.ts

```diff
@@ -154,6 +148,31 @@ export function addItem(
   });
 }
 
+function replaceEquippedItem(
+  defs: D2ManifestDefinitions | D1ManifestDefinitions,
+  item: DimItem,
+  equip: boolean
+) {
+  return produce((draftLoadout) => {
+    setEquipForItemInLoadout(defs, item, draftLoadout, equip);
+  });
+}
+
+export function dropItem(
+  defs: D2ManifestDefinitions | D1ManifestDefinitions,
+  item: DimItem,
+  equip?: boolean,
+  socketOverrides?: SocketOverrides
+): LoadoutUpdateFunction {
+  const updateFunctions: LoadoutUpdateFunction[] = [];
+  if (item.bucket.hash === BucketHashes.Subclass) {
+    updateFunctions.push(clearSubclass(defs));
+  }
+  updateFunctions.push(replaceEquippedItem(defs, item, Boolean(equip)));
+  updateFunctions.push(addItem(defs, item, equip, socketOverrides));
+  return compose(...updateFunctions);
+}
```

Comment:
Want opinions on this, rather than redefining all the logic for equipping and adding items I have composed a bunch of actions for dropping. I realise this is a bit different to current patterns, which are to extract functionality out into a helper function. Happy to move over to that if people prefer.

Reply:
Going to remove this and extract functions like we normally do so I can just get it merged

Reply:
I don't really mind it (I get what you're doing) but it veers into the "build a data structure to model logic" style of programming that can get hard to follow.

Reply:
Yeah that is why I wanted opinions. I might just extract some stuff and make it more functional.